expression/builtin_cast_vec.go

```diff
@@ -478,11 +478,44 @@ func (b *builtinCastDecimalAsDecimalSig) vecEvalDecimal(input *chunk.Chunk, resu
 }
 
 func (b *builtinCastDurationAsTimeSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinCastDurationAsTimeSig) vecEvalTime(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETDuration, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalDuration(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	result.ResizeTime(n, false)
+	result.MergeNulls(buf)
+	times := result.Times()
+	stmtCtx := b.ctx.GetSessionVars().StmtCtx
+	fsp := int8(b.tp.Decimal)
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		res, err := buf.GetDuration(i, b.tp.Decimal).ConvertToTime(stmtCtx, b.tp.Tp)
```

Comment:
Duration is a fixed-length type. Consider using https://godoc.org/github.com/pingcap/tidb/util/chunk#Column.GoDurations

Reply:
Fixed.